components/brave_vpn/browser/brave_vpn_service.cc

```diff
@@ -774,8 +774,12 @@ void BraveVpnService::SetPurchasedState(
     obs->OnPurchasedStateChanged(state, description);
 
 #if !BUILDFLAG(IS_ANDROID)
-  if (state == PurchasedState::PURCHASED)
+  if (state == PurchasedState::PURCHASED) {
     connection_api_->CheckConnection();
+
+    // Some platform needs to install services to run vpn.
+    connection_api_->InstallSystemServices();
+  }
```

Comment:
if it's not purchased we should `MaybeUninstallSystemServices`

Reply:
We can do that - originally scope was just to:
1. Delete the services for everybody. This happens during mini-installer process
2. Add the services back only if person is purchased

If we want to accommodate a 3rd state- where person had purchased it but then subscription expires, etc we can remove the services too. I'd like to suggest doing this as a follow up though as it'll add some extra complexity

Reply:
Well, I think this is still going to be an issue if people try vpn and then don't use it again, but the services are left behind. I don't think we want to go through another round of complaints over this.

Reply:
Created https://github.com/brave/brave-browser/issues/34773 so that we can do as a follow up. The big thing for this change is removing VPN services for people that have never interacted with Brave VPN in any way